core/trino-spi/src/main/java/io/trino/spi/block/Int128ArrayBlockBuilder.java

```diff
@@ -151,11 +151,11 @@ public void appendRange(ValueBlock block, int offset, int length)
         long[] rawValues = int128ArrayBlock.getRawValues();
         System.arraycopy(rawValues, (rawOffset + offset) * 2, values, positionCount * 2, length * 2);
 
-        boolean[] rawValueIsNull = int128ArrayBlock.getRawValueIsNull();
+        byte[] rawValueIsNull = int128ArrayBlock.getRawValueIsNull();
         if (rawValueIsNull != null) {
             for (int i = 0; i < length; i++) {
-                if (rawValueIsNull[rawOffset + offset + i]) {
-                    valueIsNull[positionCount + i] = true;
+                if (rawValueIsNull[rawOffset + offset + i] != 0) {
```

Comment:
== 1 ?

Reply:
We do not require the byte to be 1 as there is no way to enforce this without scanning the arrays.  So we only test zero and not zero.

Reply:
Nit: worth have `true` expressed as something different than `1` in some tests?